### PR TITLE
Synthetic Data Sampling from Manifold

### PR DESCRIPTION
**Pull request by Vivdaddy**
Hi,

I've added functionality to sample points synthetically from a manifold. These points can then be made into triplets, responded to, create embeddings on and then estimate dimensionality.

The two classes I have added are

datasets.BaseManifold - base class for manifolds

and 

datasets.LinearSubspace - a LinearSubspace embedded in higher dimensional space.

Unit tests for the LinearSubspace are in datasets.tests.test_linear_subspace.py

Please do let me know if I need to make any changes.

### CODE REVIEW

File: cblearn/datasets/_linear_subspace.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+from scipy.spatial.distance import pdist, squareform
+
+
+class LinearSubspace(BaseManifold):
+    """
+    Linear Subspace
+
+    Linear Subspace is a class for creating a hyperplane of a given
+    subspace dimension embedded in a higher dimensional space. It gives
+    a method of generating synthetic ordinal data with intrinsic
```
Comment (dekuenstle) [inline cblearn/datasets/_linear_subspace.py:15]: You write "generating synthetic ordinal data" In what sense is the generated data ordinal? Doesn't this class generate points?

File: docs/sg_execution_times.rst
Comment (dekuenstle) [inline docs/sg_execution_times.rst:1]: I think this file was committed by accident. Could you please add it to .gitignore and remove it?

File: docs/references/index.rst [hunk 2]
```diff
@@ -29,6 +29,8 @@ Loaders
     datasets.fetch_vogue_cover_similarity
     datasets.fetch_things_similarity
     datasets.fetch_similarity_matrix
+    datasets.BaseManifold
```
Comment (dekuenstle) [inline docs/references/index.rst:32]: Please do not add BaseManifold here in the public docs. It is not interesting for 99% of the users.

**Review by dekuenstle — CHANGES_REQUESTED** [review]
Hi @Vivdaddy 
thanks for the PR! Overall it looks really good! 

I don't know yet whether this should be a class or function. 
For me, the difference is: Do you typically generate points once -> function (e.g., like in sklearn make_...); do you generate points many times from the same hyperplane, and need additional methods? -> class. What do you think?

It would also be great if you could write 1-2 sentences in the docstring why this is a useful structure in our setting. Otherwise, it seems to be a bit strange to have a point sample in a comparison library.

File: docs/references/index.rst (hunk 2 above)
Comment (Vivdaddy) [reply docs/references/index.rst:32]: Will do and push in a few days

File: docs/sg_execution_times.rst
Comment (Vivdaddy) [reply docs/sg_execution_times.rst:1]: Will do and push in a few days

**Comment by dekuenstle** [comment]
Thank you for the explanation! I will merge once you pushed the agreed changes :-)

**Comment by Vivdaddy** [comment]
Yup, pushed!

**Comment by dekuenstle** [comment]
Please delete `docs/sg_execution_times.rst`, then it's ready to merge from my side.

**Pull request merged**